src/main/java/baritone/cache/CachedHomeAreas.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.cache;
+
+import baritone.api.cache.ICachedHomeAreas;
+import baritone.api.selection.ISelection;
+import baritone.api.selection.ISelectionManager;
+import baritone.api.utils.BetterBlockPos;
+import baritone.selection.Selection;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class CachedHomeAreas implements ICachedHomeAreas {
+    private Path directory;
+    CachedHomeAreas(Path directory)
+    {
+        this.directory = directory;
+        if (!Files.exists(directory)) {
+            try {
+                Files.createDirectories(directory);
+            } catch (IOException ignored) {
+            }
+        }
+
+    }
+    @Override
+    public synchronized void save(ISelectionManager manager)
+    {
+        Path fileName = this.directory.resolve("CoolShit.mp4");
```

Comment:
Don't swear where it is seen by end users, it is fine in comments, but preferably not creating the file on their computer.